From 2d92c48097cd202c8e20adfceb7bd095622978a1 Mon Sep 17 00:00:00 2001
From: "Dr. Greg Wettstein" <gw@idfusion.org>
Date: Fri, 30 Dec 2016 12:41:20 -0600
Subject: [PATCH 5/5] Implement iso-identity mapping support.

This commit adds support to main integrity processing loop to
conduct an actor/subject mapping if the IMA_MAP directive is
encountered on a subject.

With this update a full iso-identity mapping policy is available
and which represents the composite patch which this patch series
was obtained from.
---
 security/integrity/ima/ima_main.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/security/integrity/ima/ima_main.c b/security/integrity/ima/ima_main.c
index c21f09bf8b99..61624e86ecdd 100644
--- a/security/integrity/ima/ima_main.c
+++ b/security/integrity/ima/ima_main.c
@@ -242,6 +242,13 @@ static int process_measurement(struct file *file, int mask, int function,
 	if (action & IMA_AUDIT)
 		ima_audit_measurement(iint, pathname);
 
+	if (action & IMA_MAP) {
+		if (!ima_identity_is_mapped(function, iint, file, pathname)) {
+			ima_store_measurement(iint, file, pathname,
+					      xattr_value, xattr_len);
+		}
+	}
+
 out_digsig:
 	if ((mask & MAY_WRITE) && (iint->flags & IMA_DIGSIG))
 		rc = -EACCES;
-- 
2.11.0

